crates/wal-protocol/src/lib.rs

```diff
@@ -169,9 +171,9 @@ pub async fn append_envelope_to_bifrost(
     bifrost: &mut Bifrost,
     envelope: Envelope,
 ) -> Result<(LogId, Lsn), Error> {
-    let partition_id = metadata()
-        .partition_table()
-        .find_partition_id(envelope.partition_key())?;
+    let partition_table = metadata().wait_for_partition_table(Version::MIN).await?;
```

Comment:
Should this be a requirement on bifrost startup instead of yielding on every append? I'm worried about the slow creep of expensive operations in bifrost's append path.

Reply:
I can move the part which ensures the presence of the partition table to the node start-up. Then I'll add an `expect` here which encodes this contract.